service/samples/configuration_sample.js

```diff
@@ -0,0 +1,149 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+'use strict';
+
+var iothub = require('azure-iothub');
+
+var connectionString = '[IoT Hub Connection String]';
+var deviceId = '[Device ID]';
+
+var registry = iothub.Registry.fromConnectionString(connectionString);
+
+var sampleConfig = {
+  id: 'sampleconfig',
+  labels: {},
+  content: {
+    moduleContent: {
+      fakeModule: {
+        'properties.desired': {
+          prop1: 'foo'
+        }
+      }
+    }
+  },
+  targetCondition: 'tags.environment=\'prod\'',
+  priority: 20
+};
+
+var printJson = function(obj) {
+  console.log(JSON.stringify(obj, null, '  '));
+};
+
+
+var getAllConfigurations = function(done) {
```

Comment:
>getAllConfigurations [](start = 4, length = 20)

I guess these functions are done to avoid nesting? i wouldn't mind it that much.

Reply:
Since this is a sample, it's more about outlining sample scenarios.  I guess I could nest, but this seems easier to read.

---
In reply to: [176891394](https://github.com/Azure/azure-iot-sdk-node/pull/256#discussion_r176891394) [](ancestors = 176891394)